core/modules/layout/layout.admin.inc

```diff
@@ -589,9 +589,14 @@ function layout_settings_form_update_layout($form, &$form_state) {
     $layout->name = $form_state['values']['name'];
   }
   if (isset($form_state['values']['layout_template'])) {
-    $new = $form_state['values']['layout_template'];
-    $region_mapping = layout_get_legacy_region_mapping($layout->layout_template, $new);
-    $layout->setLayoutTemplate($form_state['values']['layout_template'], $region_mapping);
+    $old_template = $layout->layout_template;
+    $new_template = $form_state['values']['layout_template'];
+    if ($old_template != $new_template) {
+      $region_mapping = layout_get_legacy_region_mapping($old_template, $new_template);
+      $layout->setLayoutTemplate($new_template, $region_mapping);
```

Comment:
Perhaps, as a compromise, instead we could simply set a message here saying Backdrop detected a template change, please flash your theme cache if you just added a new template and your regions are not showing what you expect. Personally I prefer to flush caches automatically. 80% would UX benefit surely.

Reply:
Or better still, is there any way to check if there is a theme every for the new template and then we'd know for sure that this is the issue, then just explicitly message "you need to flush your theme cache" in nicer words of course.